### PR TITLE
[WIP] Add mongoose as odm

### PR DESCRIPTION
**Pull request by nicolanrizzo**

### CODE REVIEW

**Comment by cedriclombardot** [comment]
@nicolanrizzo paramconverters and basecontroller use same interfaces ?

**Comment by nicolanrizzo** [comment]
No, I don't think so. Therefore, it's a WIP. I have to check only Paramconverters and Basecontroller?

**Comment by cedriclombardot** [comment]
I think you have to check 

* ParamConverter
* BaseController
* registerModels in module

and that should be good !!

Cédric

**Comment by cedriclombardot** [comment]
@nicolanrizzo can you rebase it and add plugins

**Comment by nicolanrizzo** [comment]
@cedriclombardot  add plugins? Do you mean https://github.com/bureaux-a-partager/node-microframework/pull/8/files#diff-4904b572f1fd25f5192946fdbb38feafR93?

And rebased.

**Comment by cedriclombardot** [comment]
Yes i ‘ve merged the plugins repos

Cédric

**Comment by nicolanrizzo** [comment]
Remove plugins from constructor done @cedriclombardot

File: src/core/container.ts [hunk 1]
```diff
@@ -16,6 +15,10 @@ export class Container {
         this.services[serviceName] = service;
     }
 
+    static debugServices() {
+        console.log(Object.keys(this.services));
```
Comment (cedriclombardot) [inline src/core/container.ts]: Un dump de tout en console.table ?

Comment (cedriclombardot) [inline src/core/container.ts]: Ou genre nom et classe

**Comment by cedriclombardot** [comment]
Quid about boot before adding plugins ?

File: typings.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "lodash": "github:DefinitelyTyped/DefinitelyTyped/lodash/lodash.d.ts#d456e7ccb07a53c47bdd63957e8785402c19c579",
     "mime": "github:DefinitelyTyped/DefinitelyTyped/mime/mime.d.ts#cb5206a8ac1c9a3ddfd126f5ecea6729b2361452",
     "mocha": "github:DefinitelyTyped/DefinitelyTyped/mocha/mocha.d.ts#d6dd320291705694ba8e1a79497a908e9f5e6617",
+    "mongoose": "github:DefinitelyTyped/DefinitelyTyped/mongoose/mongoose.d.ts#5e1ca2b9ef1971a6b08d9fb6d9277f20c61887f2",
```
Comment (cedriclombardot) [inline typings.json]: no

**Comment by nicolanrizzo** [comment]
@cedriclombardot done

**Pull request merged**